template.rb

```diff
@@ -69,14 +69,24 @@
   gem 'truncate_html'
   gem 'view_component', '~> 2.46'
 
+  gem_group :test do
+    # We need to add capybara along with a javascript driver to support the provided system specs.
+    # `rails new` will add the following gems for system tests unless `--skip-test` is provided.
+    # We want to stick with them but we can't be sure about how the app was generated, so we'll
+    # add them only if they're not already in the Gemfile.
+    gem "capybara" unless Bundler.locked_gems.dependencies['capybara']
+    gem "selenium-webdriver" unless Bundler.locked_gems.dependencies['selenium-webdriver']
+    gem "webdrivers" unless Bundler.locked_gems.dependencies['webdrivers']
```

Comment:
I'd love to get rid of this hack, as that's something that end users will see. On top of that, I've experienced `Bundler.locked_gems` to be very slow sometimes.

What about modifying our spec helper with something like this for each case?

```ruby
begin
  require 'capybara'
rescue LoadError
  raise "SSF needs capybara to be available for.... Please add it to the the `:test` group in your Gemfile"
end
```

Reply:
> I'd love to get rid of this hack, as that's something that end users will see.

This won't be seen by the user, as it's happening in the app-template, that `gem` DSL is not the same as the one we have in a Gemfile. The slowness probably happens when the lock is not there already and needs to be generated (just guessing).

The proposed solution could work but given we're not dirtying anything in the user Gemfile I would consider a better experience for everything to be working out of the box (at least for the most beaten paths).

Reply:
Oh, yeah, sorry, I got confused. Yeah, if that doesn't pollute the user's Gemfile, that's the best solution 👍 Sorry for the noise.